rust/scx_utils/src/compat.rs

```diff
@@ -152,6 +155,32 @@ pub fn ksym_exists(ksym: &str) -> Result<bool> {
     Ok(tid >= 0)
 }
 
+pub fn ksym_probeable(ksym: &str) -> Result<bool> {
+    let file = std::fs::File::open("/proc/kallsyms")?;
+    let reader = std::io::BufReader::new(file);
+
+    for line in reader.lines() {
+        for sym in line.unwrap().split_whitespace() {
+            if ksym == sym {
+                return Ok(true);
+            }
+        }
+    }
```

Comment:
Not necessary right now but it'd probably be better to build a hashset and keep it in a lazy_static. Or is the worry that we wanna re-parse the whole thing in case modules have been loaded inbetween?

Reply:
I think minimally this is going to use >24MB of memory, probably not a big deal as it's swappable.

Reply:
I like this idea. Will do it in another PR though because I think it will also need a timer/expiration (or something) to periodically reset the map (for when layered starts before modules are loaded, which I think is more realistic of a use case than it sounds like due to the popularity of AMD handhelds and nvidia eGPU's (presuming eGPU's work like this on linux at all -- I've only done this w/ windows)).

Reply:
Oh, let's not overcomplicate it. Reading and discarding is fine.